pkg/controller/traceflow/controller.go

```diff
@@ -224,14 +241,28 @@ func (c *Controller) checkTraceflowStatus(tf *opsv1alpha1.Traceflow) (retry bool
 	retry = false
 	sender := false
 	receiver := false
-	for _, nodeResult := range tf.Status.Results {
-		for _, ob := range nodeResult.Observations {
+	for i, nodeResult := range tf.Status.Results {
+		for j, ob := range nodeResult.Observations {
 			if ob.Component == opsv1alpha1.SpoofGuard {
 				sender = true
 			}
 			if ob.Action == opsv1alpha1.Delivered || ob.Action == opsv1alpha1.Dropped {
 				receiver = true
 			}
+			if ob.TranslatedDstIP != "" {
+				// Add Pod ns/name to observation if TranslatedDstIP (a.k.a. Service Endpoint address) is Pod IP.
+				pods, err := c.podInformer.Informer().GetIndexer().ByIndex("podIP", ob.TranslatedDstIP)
+				if err != nil {
+					klog.Warningf("Get Pods by IP failed, error: %+v", err)
```

Comment:
Service can have external endpoints, so this might not be an error.

Reply:
Change this line to
`klog.Infof("Unable to find Pod from IP, error: %+v", err)`